app/models/cim_base_storage_extent.rb

```diff
@@ -10,10 +10,7 @@ def self._virtual_reflections
   end
 
   def self.find(*args)
-    options = args.extract_options!
-    options[:conditions] = CimStorageExtent.merge_conditions(options[:conditions], :id => base_storage_extent_ids)
-    args << options
-    CimStorageExtent.find(*args)
+    CimStorageExtent.where(:id => base_storage_ext_ids).find(*args)
```

Comment:
This may not account for other options passed via `args`, such as `:distinct => true`.  At the same time, I have no idea what the callers are doing either.  Moreover, I hate the idea of specializing AR's find method.